tests/integration/relationships/has-many-test.js

```diff
@@ -414,6 +414,64 @@ test("A hasMany updated link should not remove new children when the parent reco
   });
 });
 
+test("A hasMany updated link should not remove new children when the parent record has children already with feedback from the server after save", function(assert) {
+
+  Post.reopen({
+    comments: DS.hasMany('comment', { async: true })
+  });
+
+  Comment.reopen({
+    message: DS.belongsTo('post', { async: true })
+  });
+
+  env.adapter.createRecord = function(store, snapshot, link, relationship) {
+    return Ember.RSVP.resolve({
+      id: 5
+    });
+  };
+
+
+  env.adapter.findHasMany = function(store, snapshot, link, relationship) {
+    return Ember.RSVP.resolve([{ id: 5 }]);
+  };
+
+
+  run(function() {
+    var post = env.store.createRecord('post', {});
+    env.store.createRecord('comment', { id: 5, message: post });
+
+    post.get('comments')
+      .then(function(comments) {
+        assert.equal(comments.get('length'), 1);
+        return post.save();
+      })
+      .then(function() {
+        return env.store.push({
+          data: {
+            type: "post",
+            id: 5,
+            relationships: {
+              comments: {
+                data: [
+                  {
+                    id: 5,
+                    type: "comment"
+                  }
+                ]
+              }
+            }
+          }
+        });
```

Comment:
No it is important it showcase the bug.
On Feb 12, 2016 1:35 AM, "Brendan McLoughlin" <notifications@github.com>
wrote:

> In tests/integration/relationships/has-many-test.js
> <https://github.com/emberjs/data/pull/4154#discussion_r52691383>:
>
> > +        return env.store.push({
> > +          data: {
> > +            type: "post",
> > +            id: 5,
> > +            relationships: {
> > +              comments: {
> > +                data: [
> > +                  {
> > +                    id: 5,
> > +                    type: "comment"
> > +                  }
> > +                ]
> > +              }
> > +            }
> > +          }
> > +        });
>
> @olivierchatry <https://github.com/olivierchatry> is the store.push here
> important to this test or can we move the info about the comments
> relationship into the return form the stubbed createRecord method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/emberjs/data/pull/4154/files#r52691383>.
>